### PR TITLE
Server side render

### PR DESCRIPTION
**Pull request by StefanoPastore**
[See this comment on issue #13 ](https://github.com/ReactTraining/react-media/issues/13#issuecomment-250029997)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 lib
 umd
+node_modules
```
Comment (mjackson) [inline .gitignore]: Please don't make changes to `.gitignore`. Thanks :)

File: modules/Media.js [hunk 2]
```diff
@@ -1,38 +1,97 @@
...
+      {UnmatchElement}
+    </placeholder>)
+  }
+
   render() {
-    const { children, render } = this.props
+    const { defaultMatches, query, children, render } = this.props
     const { matches } = this.state
 
+    if (defaultMatches && this.firstRender) return this.universalRender(query, children, render)
```
Comment (mjackson) [inline modules/Media.js]: In general, this approach feels too heavy for this library. We shouldn't be rendering a `<style>` tag when rendering on the server.

Comment (StefanoPastore) [inline modules/Media.js]: I get it, but if the rendering is universal then the `<style>`tag on the server becomes meaningful as a counterpart of the client tag. Also, this is needed for crawlers which expect the markup to be the same on both sides.

Comment (mjackson) [inline modules/Media.js]: I'm sorry, but this PR is just doing way too much. There must be an easier way to achieve what you're after.

Comment (StefanoPastore) [inline modules/Media.js]: Without `<style>` tag it isn't possibile, but if you want I can change `universalRender` method as well:

```js
universalRender(){
  return MatchElement
}
```

For fix warning until we find a better solution.

What do you think?

Comment (mjackson) [inline modules/Media.js]: Well, we can't include a `<style>` tag in our output. That's taking on too much responsibility. I'm sorry, but if you'd like to take that route you'll have to do it in a separate library.

**Comment by mjackson** [comment]
I'm sorry but this PR is just too complex. My suggestion would be to just use

**Pull request closed without merging**